pkg/datapath/linux/routing/routing.go

```diff
@@ -67,7 +61,7 @@ func (info *RoutingInfo) Configure(ip net.IP, mtu int, compat bool) error {
 	var egressPriority, tableID int
 	if compat {
 		egressPriority = linux_defaults.RulePriorityEgress
-		tableID = ifindex
+		tableID = info.InterfaceNumber
```

Comment:
This will break in Azure mode because the Azure API doesn't expose the ifindex. In Azure, the `InterfaceNumber` is hardcoded to to 0. https://docs.microsoft.com/en-us/rest/api/virtualnetwork/network-interfaces/get#networkinterface